app/src/main/java/com/niesens/morsetrainer/TextSpeaker.java

```diff
@@ -70,7 +70,8 @@ public void run() {
                 }
             });
         }
-        textToSpeech.speak(text, TextToSpeech.QUEUE_FLUSH, textToSpeechParams);
+        final String speakText = (vocalize ? text : "");
+        textToSpeech.speak(speakText, TextToSpeech.QUEUE_FLUSH, textToSpeechParams);
```

Comment:
@deverac This is quite interesting.  On Android 7 (Samsung Galaxy S6) this code works just fine while on Android 9 (Samsung Galaxy S10) this causes the app to fail.  I think the textToSpeech engine notices that the text string is empty and doesn't invoke the speak operation.  This app is waiting for the speak operation to end (UtteranceProgressListener.onDone), which never happens and causes the app to get stuck.

My solution was to toggle the volume on textToSpeach:
```
textToSpeechParams.put(KEY_PARAM_VOLUME, vocalize ? "1" : "0");
textToSpeech.speak(text, TextToSpeech.QUEUE_FLUSH, textToSpeechParams);
```